plugins/coordinator/plugin.go

```diff
@@ -152,7 +153,15 @@ func initCoordinator(bootstrap bool, startIndex uint32, powHandler *powpackage.H
 		}
 	}
 
-	inMemoryEd25519MilestoneSignerProvider := coordinator.NewInMemoryEd25519MilestoneSignerProvider(privateKeys, deps.Storage.KeyManager(), deps.NodeConfig.Int(protocfg.CfgProtocolMilestonePublicKeyCount))
+	var signingProvider coordinator.MilestoneSignerProvider
+	switch deps.NodeConfig.String(CfgCoordinatorSigningProvider) {
```

Comment:
This switch should happen sooner: you are still trying to load private keys from `COO_PRV_KEYS` even is signerProvider is set to `remote`!!